creator/files/views.py

```diff
@@ -46,11 +48,18 @@ def download(request, study_id, file_id, version_id=None):
             # same file/object information, but will not consider it for now
             download_token = None
 
+        # Try to resolve a dev download token
+        try:
+            dev_token = DevDownloadToken.objects.get(token=token)
+        except DevDownloadToken.DoesNotExist:
+            dev_token = None
+
     if ((user is None                           # No user present
```

Comment:
Phew, this is confusing to me but I think it is saying:

```python
authenticated = user is not None and user.is_authenticated
user_authorized = (study_id in user.ego_groups or ADMIN) and download_token
developer_authorized = ADMIN and dev_token

if not (authenticated and (user_authorized or developer_authorized))
    return HttpResponseNotFound('Not authenticated to download the file.')
```

Reply:
This isn't necessarily a request for change, just trying to make sure I understand that logic correctly.

Reply:
heh yes you can tell I thought that too by all the comments. Yes that logic looks correct.